pkg/utils/cel/quantity_test.go

```diff
@@ -0,0 +1,284 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cel
+
+import (
+	"maps"
+	"math"
+	"slices"
+	"testing"
+
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func TestQuantityCalculation(t *testing.T) {
+	tests := []struct {
+		name       string
+		expression string
+		expected   float64
+	}{
+		{
+			name:       "parse tebibyte quantity",
+			expression: `Quantity("1Ti")`,
+			expected:   1099511627776,
+		},
+		{
+			name:       "add tebibyte",
+			expression: `Quantity("1Ti") + Quantity("1Ti")`,
+			expected:   2199023255552,
```

Comment:
Use readable expected value
```suggestion
			expected:   `Quantity("2Ti")`,
```

Reply:
Unfortunately, achieving this will likely require even more effort on my part. 

> [!note] 
> [Quantity](https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/quantity/)
> When a Quantity is parsed from a string, it will remember the type of suffix it had, and will use the same type again when it is serialized.

In Kwok’s implementation, the Add and Subtract functions reuse the official library's methods, allowing them to retain the suffix upon deserialization. 

However, for multiplication and division, custom implementations are used, and this process loses the suffix information. As a result, when testing `Quantity("1Ti") * 2`, calling `String()` on the Quantity outputs `2199023255552` rather than `2Ti`. 

Do you have any suggestions on this situation?

Reply:
Convert the expected value to milli, then compare it.